cherrybomb 1.0.1 (new formula)

DeliciousBounty opened the pull request:
<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

chenrui333 commented on Formula/c/cherrybomb.rb:
```diff
@@ -0,0 +1,15 @@
+  sha256 "da863e4e58fdde80d64ed91e564fe032792063504030a12ee3f86b6fc60e108f"
+  license "Apache 2.0"	
+  def install 
+    bin.install "cherrybomb" # Assuming this is the binary or executable
```
need to build from source

chenrui333 commented on Formula/c/cherrybomb.rb:
```diff
@@ -0,0 +1,15 @@
+  end
+
+  test do
+    output = shell_output("#{bin}/cherrybomb --version")
+    assert_match "1.0.1", output
```
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

chenrui333 commented on Formula/c/cherrybomb.rb:
```diff
@@ -0,0 +1,15 @@
+  homepage "https://blstsecurity.com"
+  url "https://github.com/blst-security/cherrybomb/releases/download/v1.0.1/cherrybomb"
+  sha256 "da863e4e58fdde80d64ed91e564fe032792063504030a12ee3f86b6fc60e108f"
+  license "Apache 2.0"	
```
```suggestion
  license "Apache 2.0"

```

p-linnane commented:
@DeliciousBounty I've rewritten this to build from source, and updated the test. Please add an additional test that verifies some functionality of the app. Checking the version is good, but it's not enough of a functionality test to stand on its own.

DeliciousBounty commented:
> @DeliciousBounty I've rewritten this to build from source, and updated the test. Please add an additional test that verifies some functionality of the app. Checking the version is good, but it's not enough of a functionality test to stand on its own.
Thank you for your reply, I will add the functionality test

DeliciousBounty commented:
I would happy have some help for the "commit style" check thank for advance

chenrui333 commented on Formula/c/cherrybomb.rb:
```diff
@@ -0,0 +1,26 @@
+  end
+
+  test do
+    file_url = "https://raw.githubusercontent.com/blst-security/cherrybomb/main/images/api-with-examples.yaml"
```
if this is part of source tarball, you can just do `pkgshare.install xxx` and it would be avail for the test block

chenrui333 commented on Formula/c/cherrybomb.rb:
```diff
@@ -0,0 +1,26 @@
+    assert_match "active", output
+  end
+
+  def curl_output(url, to:)
+    system "curl", "-o", to, "-L", url
+  end
```
seems a bit unnecessary, checkout other `curl -s` examples in the codebase

chenrui333 commented:
@DeliciousBounty, thanks for your first contribution to homebrew-core! 🎉

DeliciousBounty commented:
> @DeliciousBounty, thanks for your first contribution to homebrew-core! 🎉Hey Thanks for your Help!
